CMakeLists.txt

```diff
@@ -42,8 +38,16 @@ endif()
 
 add_subdirectory(extern/pybind11)
 pybind11_add_module(${PROJECT_NAME} MODULE src/simulator.cpp)
+
+if(NOT EXISTS "${PROJECT_SOURCE_DIR}/extern/rapidjson/include")
+    message(FATAL_ERROR "The submodules were not downloaded! GIT_SUBMODULE was turned off or failed. Please update submodules and try again.")
+endif()
+include_directories(extern/rapidjson/include)
+
+add_subdirectory(src)
```

Comment:
why do you put `add_subdirectory(src)` here?
does is still work since you have put this line after 
`pybind11_add_module(${PROJECT_NAME} MODULE src/simulator.cpp)`